sdk/spring/spring-cloud-azure-feature-management/src/test/java/com/azure/spring/cloud/feature/management/ValidationsTest.java

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.spring.cloud.feature.management;
+
+import com.azure.spring.cloud.feature.management.filters.TargetingFilter;
+import com.azure.spring.cloud.feature.management.filters.TargetingFilterTestContextAccessor;
+import com.azure.spring.cloud.feature.management.filters.TimeWindowFilter;
+import com.azure.spring.cloud.feature.management.implementation.FeatureManagementConfigProperties;
+import com.azure.spring.cloud.feature.management.implementation.FeatureManagementProperties;
+import com.azure.spring.cloud.feature.management.validationstests.models.ValidationTestCase;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.MapperFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.json.JsonMapper;
+import com.fasterxml.jackson.databind.type.CollectionType;
+import com.fasterxml.jackson.databind.type.TypeFactory;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.springframework.context.ApplicationContext;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.LinkedHashMap;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(SpringExtension.class)
+public class ValidationsTest {
+    @Mock
+    private ApplicationContext context;
+    @Mock
+    private FeatureManagementConfigProperties configProperties;
+
+    private final ObjectMapper objectMapper = JsonMapper.builder()
+        .configure(MapperFeature.ACCEPT_CASE_INSENSITIVE_PROPERTIES, true).build();
+    private final String testCaseFolderPath = "validations-tests";
+    private final String inputsUser = "user";
+    private final String inputsGroups = "groups";
+    private final String sampleFileNameFilter = "sample";
+    private final String testsFileNameFilter = "tests";
+
+    @BeforeEach
+    public void setup() {
+        MockitoAnnotations.openMocks(this);
+        when(configProperties.isFailFast()).thenReturn(true);
+        when(context.getBean(Mockito.contains("TimeWindow"))).thenReturn(new TimeWindowFilter());
```

Comment:
We shouldn't need to mock the `TimeWindowFilter`, they were written in a way that should always result in it being true. We aren't testing edge cases with the validation tests. The dates that need to be in the future are all in the year `3023` and all the ones in the past are the year `2023`. These tests without a mock will run fine for another 999 or so years.

Reply:
If I didn't mock the TimeWindowFilter, then the feature filter will be null when run the following code:
```
FeatureFilter featureFilter = (FeatureFilter) context.getBean(filter.getName());
```

I just add this mock to make sure it will return a time window filter instance.